docs/source/configuration.rst

```diff
@@ -0,0 +1,201 @@
+Configuration
+=============
+
+Taking full advantage of Dask sometimes requires user configuration.
+This might be to control logging verbosity, specify cluster configuration,
+provide credentials for security, or any of several other options that arise in
+production.
+
+Configuration is specified in one of the following ways:
+
+1.  YAML files in ``~/.config/dask/`` or ``/etc/dask/``
+2.  Environment variables like ``DASK_SCHEDULER__WORK_STEALING=True``
+3.  Default settings within sub-libraries
+
+This combination makes it easy to specify configuration in a variety of
+settings ranging from personal workstations, to IT-mandated configuration, to
+docker images.
+
+
+Access Configuration
+--------------------
+
+.. currentmodule:: dask
+
+.. autosummary::
+   dask.config.get
+
+Configuration is usually read by using the ``dask.config`` module, either with
+the ``config`` dictionary or the ``get`` function:
+
+.. code-block:: python
+
+   >>> import dask
+   >>> import dask.distributed  # populate config with distributed defaults
+   >>> dask.config.config
+   {
+     'logging': {
+       'distributed': 'info',
+       'bokeh': 'critical',
+       'tornado': 'critical',
+     }
+     'admin': {
+       'log-format': '%(name)s - %(levelname)s - %(message)s'
+     }
+   }
+
+   >>> dask.config.get('logging')
+   {'distributed': 'info',
+    'bokeh': 'critical',
+    'tornado': 'critical'}
+
+   >>> dask.config.get('logging.bokeh')  # use `.` for nested access
+   'critical'
+
+You may wish to inspect the ``dask.config.config`` dictionary to get a sense
+for what configuration is being used by your current system.
+
+
+Specify Configuration
+---------------------
+
+YAML files
+~~~~~~~~~~
+
+It is common to specify configuration in YAML files any of a few possible
+locations:
+
+1.  The ``~/.config/dask`` directory in the user's home directory
+2.  The ``{sys.prefix}/etc/dask`` directory local to Python
+3.  The root ``/etc/dask/`` directory
+
+Dask searches for *all* YAML files within each of these directories and merges
+them together, preferring configuration files closer to the user over system
+configuration files (preference follows the order in the list above).
+Additionally users can specify a path with the ``DASK_CONFIG`` environment
+variable, that takes precedence at the top of the list above.
+
+The contents of these YAML files are merged together, allowing different
+dask subprojects like ``dask-kubernetes`` or ``dask-ml`` to manage configuration
+files separately, but have them merge into the same global configuration.
+
+*Note: for historical reasons we also look in the ``~/.dask`` directory for
+config files.  This is deprecated and will soon be removed.*
+
```

Comment:
Maybe it would be worth showing a snippet from yaml file. You could use scheduler.work_stealing and scheduler.allowed_failures, i.e. the same example as environment variables below or something more complete as https://distributed.readthedocs.io/en/latest/configuration.html#user-wide-configuration, not sure.

Reply:
Good idea, I've added an example yaml file snippet